templates/newsroom/newsroom.js

```diff
@@ -0,0 +1,393 @@
+  const table = doc.querySelector('.table table') || doc.querySelector('table');
+  if (!table) return;
+
+  const tbody = table.querySelector('tbody');
+  if (!tbody) return;
+
+  // Get all rows as an array
+  const rows = Array.from(tbody.querySelectorAll('tr'));
+
+  // Filter out hidden rows (from current filters) and store their visibility
+  const rowsWithVisibility = rows.map((row) => ({
+    element: row,
+    isVisible: row.style.display !== 'none',
+    firstCellText: row.querySelector('td:first-child')?.textContent.trim() || '',
+  }));
+
+  // Sort the array based on first column text
+  rowsWithVisibility.sort((rowA, rowB) => {
+    const textA = rowA.firstCellText.toLowerCase();
+    const textB = rowB.firstCellText.toLowerCase();
+    if (order === 'asc') {
+      return textA.localeCompare(textB);
+    }
+    return textB.localeCompare(textA);
+  });
+
+  // Clear the tbody
+  tbody.innerHTML = '';
+
+  // Re-append rows in sorted order, maintaining their visibility
+  rowsWithVisibility.forEach(({ element, isVisible }) => {
+    // Restore visibility state
+    element.style.display = isVisible ? '' : 'none';
+    tbody.appendChild(element);
+  });
+
+  // Add visual feedback to show which sort is active
+  updateSortButtonStates(doc, order);
+}
+
+function setupSortButtons(doc) {
+  const filterTopBtn = doc.querySelector('.filter-top-btn');
+  const filterBottomBtn = doc.querySelector('.filter-bottom-btn');
+
+  if (!filterTopBtn || !filterBottomBtn) {
+    return;
+  }
+
+  // Sort ascending (A-Z) when top button is clicked
+  filterTopBtn.addEventListener('click', (e) => {
+    e.preventDefault();
+    sortTableByFirstColumn(doc, 'asc');
+  });
+
+  // Sort descending (Z-A) when bottom button is clicked
+  filterBottomBtn.addEventListener('click', (e) => {
+    e.preventDefault();
+    sortTableByFirstColumn(doc, 'desc');
+  });
+}
+
+export default async function decorate(doc) {
+  const $main = doc.querySelector('main');
+  const $section = section();
+  const $filterContainer = div({ class: 'media-filter-container' });
+
+  // Fetch placeholders from JSON endpoint
+  const placeholders = await fetchPlaceholders();
+
+  // Get placeholder text with fallbacks using the correct keys from your JSON
+  const typePlaceholder = getPlaceholderText(placeholders, 'media-newsroom-type-filter', 'Type');
+  const viewAllText = getPlaceholderText(placeholders, 'media-newsroom-view-filter', 'View all');
+  const searchPlaceholder = getPlaceholderText(placeholders, 'media-newsroom-search-filter', 'Search...');
+
+  // Right container for filter toggle buttons
+  const $mediaFilterRight = div({ class: 'media-filter-container-right' });
+  const $filterTopBtn = a({ class: 'filter-top-btn' });
+  const $filterBottomBtn = a({ class: 'filter-bottom-btn' });
+
+  $mediaFilterRight.append($filterTopBtn, $filterBottomBtn);
+
+  // Left container for filter controls
+  const $mediaFilterLeft = div(
```

Comment:
seems you have already referred placeholders from json, so this should be the way for the final logic.
Suggest to please remove unrequired extra functions from this file which are repeating similar logic and make this code a bit optimized in that sense.